### PR TITLE
[GH-1011] Added conditional to avoid posting multiple notifications in case of duplicate subscriptions

### LINKED ISSUE #1011: The user is getting multiple notifications in the channel or DM/GM on MM for a event on Jira when duplicate subscriptions are created.
### Summary
The user can create duplicate subscription for an existing subscription and it results in generating same notification the number of times the subscription is duplicated.
### Steps to reproduce
1. Create a subscription using the `/jira subscribe edit` slash command in any desired channel or DM/GM on MM.
2. Create a duplicate subscription for the above added subscription using the `/jira subscribe edit` slash command in the same desired channel or DM/GM on MM.
3. Create the event on Jira for which the above subscriptions were created in the desired channel or DM/GM on MM.
### Expected behavior
The user should get only one notification for the event in the desired channel or DM/GM on MM.
### Actual behavior
The user is getting N number of notifications for the event in the desired channel or DM/GM where N is the number of subscriptions created for that event on MM.
### Screenshots
image

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
#### Summary
- Added conditional to avoid posting multiple notifications in case of duplicate subscriptions

#### What to test:
- Not getting multiple notifications in a channel in case of duplicate subscriptions.

#### How to test:
1. Connect you Jira account.
2. Create multiple subscription fora same event.
3. Trigger the desired event from Jira.

#### Ticket Link
Fixes  #1011

### CODE REVIEW

File: server/subscribe.go [hunk 1]
```diff
@@ -209,7 +209,9 @@ func (p *Plugin) getChannelsSubscribed(wh *webhook, instanceID types.ID) ([]Chan
 	subIds := subs.Channel.ByID
 	for _, sub := range subIds {
 		if p.matchesSubsciptionFilters(wh, sub.Filters) {
-			channelSubscriptions = append(channelSubscriptions, sub)
+			if !HasSubscription(sub, channelSubscriptions) {
```
Comment (wiggin77) [inline server/subscribe.go:212]: Why not use a map for avoid duplicates instead of walking the list for every append?

Comment (Kshitij-Katiyar) [reply server/subscribe.go]: Hey @wiggin77, I have implemented the suggestion. Please have a look.

File: server/subscribe.go [hunk 2]
```diff
@@ -206,10 +206,13 @@ func (p *Plugin) getChannelsSubscribed(wh *webhook, instanceID types.ID) ([]Chan
 	}
 
 	var channelSubscriptions []ChannelSubscription
+	subscriptionMap := make(map[string]bool)
 	subIds := subs.Channel.ByID
 	for _, sub := range subIds {
 		if p.matchesSubsciptionFilters(wh, sub.Filters) {
-			channelSubscriptions = append(channelSubscriptions, sub)
+			if !subscriptionMap[sub.ChannelID] {
+				channelSubscriptions = append(channelSubscriptions, sub)
```
Comment (wiggin77) [inline server/subscribe.go:214]: You need to add it to the map for it to avoid collisions.

**Review by wiggin77 — CHANGES_REQUESTED** [review]

File: server/subscribe.go (hunk 2 above)
Comment (Kshitij-Katiyar) [reply server/subscribe.go:215]: @wiggin77 updated the map for every iteration. Please have a look

**Review by wiggin77 — APPROVED** [review]

**Review by AayushChaudhary0001 — APPROVED** [review]
This PR has been tested for the following scenarios:-
1. Multiple subscriptions for the same event
2. Duplicate subscription for the same event

The PR is working fine for the above conditions, LGTM. Approved

**Pull request merged**